contracts/implementations/Mapper.sol

```diff
@@ -1,11 +1,40 @@
 //SPDX-License-Identifier: BUSL-1.1
 pragma solidity ^0.8.13;
 
+import "../interfaces/INFT.sol";
 import "../libraries/MapperStorage.sol";
 
 /// @title Mapper
-/// @notice Contract to map relationships between nodes
+/// @notice Contract to map relationships between nodes and other contracts
 contract Mapper {
+    event BeneficiarySet(
+        address indexed idProxyAddress,
+        uint256 nodeId,
+        address indexed beneficiary
+    );
+
+    /// @notice Sets the beneficiary associated with the pair idProxy/nodeId
+    /// @dev Only the nodeId owner can set a beneficiary
+    /// @param idProxyAddress The address of the NFT proxy
+    /// @param nodeId The node Id to be associated with the beneficiary
+    /// @param beneficiary The address to be a beneficiary
+    function setBeneficiary(
```

Comment:
Do we want to allow people to call this for _any_ NFT, even ones not under the DIMO umbrella? It strikes me as odd.

Reply:
I haven't gotten used to letting users perform transactions yet. Fixed